### ISSUE TITLE
[Core] Awaiting a function stops rendering

### ISSUE THREAD

**Issue by BlueStaggo** [ChrisDill/Raylib-cs#116]
Before submitting a new issue, please verify and check:

 - [x] The issue is specific to Raylib-cs and not raylib
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of Raylib-cs

### Issue description

I wanted to add asynchronous functionality into my game. However, even when I await an empty task, Raylib freezes up. I know it is just Raylib because other code like printing to the console works.

### Environment

Operating System: Windows 11 22000
GPU: NVidia GTX 1650
CPU: Intel i5-10400F
RAM: 16GB
.NET version: .NET 6
Raylib-cs version: 4.0.0.1

### Issue screenshot

Raylib seems to freeze up (cursor is drawn on)...
![image](https://user-images.githubusercontent.com/71090710/182036239-edeeabd4-ae4d-4df8-a637-bd77e38eab43.png)
...yet other code works just fine.
![image](https://user-images.githubusercontent.com/71090710/182036194-9332e888-355b-41e2-85a0-d81af1ffcef0.png)

### Code example

```cs
using Raylib_cs;

Raylib.InitWindow(640, 480, "Raylib");
Raylib.SetTargetFPS(60);

await Task.Run(() => Console.WriteLine("Oh hey there!"));

Console.WriteLine("It tries to draw");
while (!Raylib.WindowShouldClose())
{
    Raylib.BeginDrawing();
    {
        Raylib.ClearBackground(Color.BEIGE);
        Raylib.DrawText("You can't\nsee me!", 8, 8, 80, Color.WHITE);
        Console.WriteLine("Bing");
    }
    Raylib.EndDrawing();
}
Console.WriteLine("It's given up on drawing");

Raylib.CloseWindow();
```

**Comment by BlueStaggo** [comment]
Turns out that I do not know much about `await` and without `await` it works just fine.

**Comment by ChrisDill** [comment]
@BlueStaggo I get the same issue using the code example. Using await here makes the top level statement async which causes issues since drawing must happen on the same thread that called `InitWindow`.

If the task needs to finish before the loop you can wait for it manually. If not then move it into a separate function that is not async and call that from your example.
